oxo.py

```diff
@@ -1,29 +1,98 @@
+        :param y: grid y position
+        :return: current contents of grid at specified position
+        ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+        """
+
+        return self.squares[y][x]
+
+    def set_square(self, x, y, player):
+        """
+        _______________________________________________________________________
+         If the specified square is currently empty (0), fill it with mark and
+         return True. Otherwise , leave it as-is and return False.
+        :param x: grid x position
+        :param y: grid y position
+        :param player: current player, also used for setting the
+                                    square's contents (0, 1, or 2)
+        :return: True or False depending on whether square is full
+        ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+        """
+        self.current_player = player
 
-    def set_square(self, x, y, mark):
-        """ If the specified square is currently empty (0), fill it with mark and return True.
-            If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        if self.squares[y][x] == 0:
+            self.squares[y][x] = player
+            return True
+        else:
+            return False
 
     def is_board_full(self):
-        """ If there are still empty squares on the board, return False.
-            If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        """
+        _______________________________________________________________________
+        If there are still empty squares on the board, return False.
+            If there are no empty squares, return True.
+        :return: True or False depending on whether all squares are full
+        ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+        """
+
+        if 0 in [items for sublist in self.squares for items in sublist]:
+            return False
+
+        else:
+            return True
 
     def get_winner(self):
-        """ If a player has three in a row, return 1 or 2 depending on which player.
-            Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+        """
+        _______________________________________________________________________
+        If a player has three in a row, return 1 or 2 depending on which
+        player. Otherwise, return 0.
+        :param player: current player (1 or 2)
+        :return: winning player (0 if no win this turn)
+        ¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
+        """
+
+        for i in range(0, len(self.squares)):
+            # check horizontal win
+            if self.squares[i][0] == \
+                    self.squares[i][1] == \
+                    self.squares[i][2] != 0:
+                return self.current_player
```

Comment:
What if a player other than `self.current_player` has made a line? This won't happen during normal play or in the unit tests, but could be a potential bug